fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -2480,14 +2517,20 @@ public static String buildSearchSelfUri(String requestUriString, FHIRSearchConte
      *
      * @param message
      *     The error message.
-     * @param lenient
-     *     A flag indicating lenient or strict mode.
+     * @param issueType
+     *     The issue type.
+     * @param context
+     *     The search context.
+     * @param alwaysThrow
+     *     Determine whether exception should be thrown even if in lenient mode.
      * @throws FHIRSearchException
      */
-    private static void manageException(String message, boolean lenient) throws FHIRSearchException {
-        if (lenient) {
+    private static void manageException(String message, IssueType issueType, FHIRSearchContext context, boolean alwaysThrow) throws FHIRSearchException {
```

Comment:
I don't love the `alwaysThrow` flag, but I guess if we need the method to behave differently in different contexts, then I suppose its need

Reply:
Yeah, I couldn't think of a better way to keep this commonly used method in place, but still get both types of behavior.